### ISSUE TITLE
I have the following problem with Grad cam performing semantic segmentation :An exception occurred in CAM with block: <class 'IndexError'>. Message: index 1 is out of bounds for dimension 0 with size 1

### ISSUE THREAD

**Issue by Chris871029nose** [jacobgil/pytorch-grad-cam#447]
import warnings
warnings.filterwarnings('ignore')
warnings.simplefilter('ignore')
from torchvision.models.segmentation import deeplabv3_resnet50
import torch
import torch.functional as F
import numpy as np
import requests
import torchvision
from PIL import Image
from pytorch_grad_cam.utils.image import show_cam_on_image, preprocess_image
import cv2
import models

image_path = "./examples/03630.jpg"
image = cv2.imread(image_path, 1)[:, :, ::-1]
image = image[240: , :]
image = cv2.resize(image, (976,208), interpolation=cv2.INTER_LINEAR)
rgb_img = np.float32(image) / 255
input_tensor = preprocess_image(rgb_img,
                                mean=[0.485, 0.456, 0.406],
                                std=[0.229, 0.224, 0.225])
# Taken from the torchvision tutorial
# https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html
num_class = 5 
model = models.ERFNet(num_class)
model = torch.nn.DataParallel(model).cuda()

checkpoint = torch.load('ERFNet_trained.tar')
torch.nn.Module.load_state_dict(model, checkpoint['state_dict'])

for name, layer in model.module.decoder.output_conv.named_children():
    print(name)

model = model.eval()

if torch.cuda.is_available():
    model = model.cuda()
    input_tensor = input_tensor.cuda()

output, output_exist = model(input_tensor)
print(type(output), type(output_exist))

normalized_masks = torch.nn.functional.softmax(output, dim=1).cpu()
sem_classes = [
   'background', 'lane1', 'lane2', 'lane3', 'lane4' 
]

sem_class_to_idx = {cls: idx for (idx, cls) in enumerate(sem_classes)}
lane1_category = sem_class_to_idx["lane1"]
lane1_mask = normalized_masks[0, :, :, :].argmax(axis=0).detach().cpu().numpy()
print(lane1_mask.shape)
lane1_mask_uint8 = 255 * np.uint8(lane1_mask > 0.5)
lane1_mask_float = np.float32(lane1_mask == lane1_category)
print(lane1_mask_float)
both_images = np.hstack((image, np.repeat(lane1_mask_uint8[:, :, None], 3, axis=-1)))
b_image = Image.fromarray(both_images)
b_image.save('test.jpg')

from pytorch_grad_cam import GradCAM
class SemanticSegmentationTarget:
    def __init__(self, category, mask):
        self.category = category
        self.mask = torch.from_numpy(mask)
        if torch.cuda.is_available():
            self.mask = self.mask.cuda()
        
    def __call__(self, model_output):
        return (model_output[self.category, :, : ] * self.mask).sum()

    
target_layers = [model.module.decoder.output_conv]
targets = [SemanticSegmentationTarget(lane1_category, lane1_mask_float)]
with GradCAM(model=model,
             target_layers=target_layers,
             use_cuda=torch.cuda.is_available()) as cam:
    grayscale_cam = cam(input_tensor=input_tensor,
                        targets=targets)[0, :]
    cam_image = show_cam_on_image(rgb_img, grayscale_cam, use_rgb=True)

    c_image = Image.fromarray(cam_image)
    c_image.save('c_test.jpg')
    
 The above is my code. My model will output 5 categories, but I can only execute the results of the background category. If I execute other categories, the above error will occur.
